File: ccc00s4.cpp
#include <iostream>

using namespace std;

int n, d, g, c;

int * clubs;

int * k;

int main(){
    cin >> d;
    cin >> g;
    k = new int [d+1];
    for (int x = 0; x<d+1; x++){
        k[x] = 0;
    }
    clubs = new int [g];
    for (int x = 0; x<g; x++){
        cin >> c;
        clubs[x] = c;
    }
    for (int x = 1; x<d+1; x++){
        for (int y = 0; y<g; y++){
            if (x-clubs[y] >= 0 and k[x-clubs[y]] != -1) {
                int outcome = k[x-clubs[y]]+1;
                if (outcome < k[x] or k[x] == 0){
                    k[x] = outcome;
                }
            }
        }
        if (k[x] == 0){
            k[x] = -1;
        }
    }
    if (k[d] == -1){
        cout << "Roberta acknowledges defeat.";
    } else {
        cout << "Roberta wins in " << k[d] << " strokes.";
    }
    return 0;
}

File: ccc00s4.py
import sys

inp = sys.stdin.readlines()

d = [0 for x in range(int(inp.pop(0).strip('\n'))+1)]

g = int(inp.pop(0).strip('\n'))

clubs = []

for x in range(g):
    clubs.append(int(inp.pop(0).strip('\n')))

for x in range(1, len(d)):
    for y in range(len(clubs)):
        if x-clubs[y] >= 0 and d[x-clubs[y]] != -1:
            outcome = d[x-clubs[y]]+1
            if outcome < d[x] or d[x] == 0:
                d[x] = outcome
    if d[x] == 0:
        d[x] = -1
            
            
if d[-1] == -1:
    print('Roberta acknowledges defeat.')
else:
    print(f'Roberta wins in {d[-1]} strokes.')